glib/src/unichar.rs

```diff
@@ -0,0 +1,239 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::{
+    translate::{FromGlib, IntoGlib},
+    UnicodeBreakType, UnicodeScript, UnicodeType,
+};
+
+mod sealed {
+    pub trait Sealed {}
+    impl Sealed for char {}
+}
+
+// rustdoc-stripper-ignore-next
+/// The kind of decomposition to perform
+#[derive(Clone, Copy)]
+pub enum DecompositionKind {
+    // rustdoc-stripper-ignore-next
+    /// Compatbility decomposition
+    Compatibility,
+
+    // rustdoc-stripper-ignore-next
+    /// Canonical decomposition
+    Canonical,
+}
+
+impl DecompositionKind {
+    fn into_bool(self) -> bool {
+        use DecompositionKind::*;
+        match self {
+            Compatibility => true,
+            Canonical => false,
+        }
+    }
+}
+
+// rustdoc-stripper-ignore-next
+/// The result of a single step of the Unicode canonical decomposition algorithm
+pub enum DecompositionResult {
+    // rustdoc-stripper-ignore-next
+    /// The character could not be decomposed further
+    NoDecomposition,
+    // rustdoc-stripper-ignore-next
+    // A 'singleton' decomposition, which means the character was replaced by another
+    Singleton(char),
+    // rustdoc-stripper-ignore-next
+    /// The first character may decompose further, but the second cannot
+    Pair(char, char),
+}
+
+// rustdoc-stripper-ignore-next
+/// This trait provides access to Unicode character classification and manipulations functions
+/// provided by GLib that do not exist in the standard library
+#[doc(alias = "g_unichar")]
+pub trait Unichar: sealed::Sealed + Copy {
+    #[doc(alias = "g_unichar_type")]
+    #[doc(alias = "unichar_type")]
+    fn unicode_type(self) -> UnicodeType;
+
+    #[doc(alias = "g_unichar_break_type")]
+    #[doc(alias = "unichar_break_type")]
+    fn break_type(self) -> UnicodeBreakType;
+
+    #[doc(alias = "g_unichar_get_script")]
+    #[doc(alias = "unichar_get_script")]
+    fn script(self) -> UnicodeScript;
+
+    #[doc(alias = "g_unichar_combining_class")]
+    #[doc(alias = "unichar_combining_class")]
+    // There's no enum defining named constants for this one as far as I can tell
+    fn combining_class(self) -> i32;
```

Comment:
You could create one :)

Reply:
There's quite a few of these ([UAX #44](https://www.unicode.org/reports/tr44/)], section 5.7.4 talks about them, [PropertyValueAliases.txt](https://www.unicode.org/Public/15.0.0/ucd/PropertyValueAliases.txt) (search for `# Canonical_Combining_Class (ccc)`) has a more complete list). Should we have an enum that has separate variants for the named ones, with the unnamed ones going into a generic `Ccc` variant? The gtk demos seem to assign more meaning to some combining classes but I'm not sure where they get that from. (see https://github.com/GNOME/gtk/blob/fa4ad1590dbbcaf1a0eb230d2a20d0a867f8b752/demos/gtk-demo/unicode-names.c#L101)

Reply:
Not sure we should do that. We would have to keep that enum up to date :) I wonder why this is not already done in GLib though...

Let's keep it a plain integer then for now (but maybe it should be unsigned?)

Reply:
I think the best type for this might be u8?

Reply:
If it's only a byte and can't be more than that, sure